### PR TITLE
add methods to BoxedResultSet for getting values

### PR DESCRIPTION
**Pull request by vechis**
Integer, Boolean, Timestamp, Date, DateString

### CODE REVIEW

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 1]
```diff
@@ -1,10 +1,37 @@
 package ru.spb.devclub.utils.db;
 
+import java.sql.Date;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.text.DateFormat;
 
 public interface BoxedResultSet extends ResultSet {
+    <T extends Enum<T>> T getEnumValue(String columnName, T[] enumItems) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+
+    <T extends Enum<T>> T getEnumValue(int columnIndex, T[] enumItems) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
+
+    Integer getIntOrNull(String columnName) throws SQLException;
+
+    Integer getIntOrNull(int columnIndex) throws SQLException;
+
+    Boolean getBooleanOrNull(String columnName) throws SQLException;
+
+    Boolean getBooleanOrNull(int columnIndex) throws SQLException;
+
+    Timestamp getTimestampOrNull(String columnName) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 4]
```diff
@@ -1,10 +1,37 @@
...
+
+    Timestamp getTimestampOrNull(int columnIndex) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 5]
```diff
@@ -1,10 +1,37 @@
...
+
+    Date getDateOrNull(String columnName) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 6]
```diff
@@ -1,10 +1,37 @@
...
+
+    Date getDateOrNull(int columnIndex) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 7]
```diff
@@ -1,10 +1,37 @@
...
+
+    String getDateStringOrNull(String columnName, DateFormat dateFormat) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java [hunk 8]
```diff
@@ -1,10 +1,37 @@
...
+
+    String getDateStringOrNull(int columnIndex, DateFormat dateFormat) throws SQLException;
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: **The Single Responsibility Principle.**
```suggestion
```

File: src/main/java/ru/spb/devclub/utils/db/SimpleBoxedResultSet.java [hunk 9]
```diff
@@ -1,23 +1,84 @@
...
+    }
+
+    @Override
+    public String getDateStringOrNull(String columnName, DateFormat dateFormat) throws SQLException {
+        return ResultSetUtils.getDateString(resultSet, columnName, dateFormat);
+    }
+
+    @Override
+    public String getDateStringOrNull(int columnIndex, DateFormat dateFormat) throws SQLException {
+        return ResultSetUtils.getDateString(resultSet, columnIndex, dateFormat);
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/SimpleBoxedResultSet.java]: **The Single Responsibility Principle.**

File: src/main/java/ru/spb/devclub/utils/db/SimpleBoxedResultSet.java [hunk 10]
```diff
@@ -1,23 +1,84 @@
...
+    @Override
+    public <T extends Enum<T>> T getEnumValue(String columnName, T[] enumItems) throws SQLException {
+        return ResultSetUtils.getEnumValue(resultSet, columnName, enumItems);
+    }
+
+    @Override
+    public <T extends Enum<T>> T getEnumValue(int columnIndex, T[] enumItems) throws SQLException {
+        return ResultSetUtils.getEnumValue(resultSet, columnIndex, enumItems);
+    }
+
```
Comment (alexengrig) [inline src/main/java/ru/spb/devclub/utils/db/SimpleBoxedResultSet.java]: **The Single Responsibility Principle.**

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 1 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 2 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 3 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 4 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 5 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 6 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 7 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

File: src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java (hunk 8 above)
Comment (vechis) [reply src/main/java/ru/spb/devclub/utils/db/BoxedResultSet.java]: remove

**Pull request merged**